Remove Embedly from tox

tomkins opened the pull request:
Personally I prefer running my tests inside tox, however currently Embedly is a bit of a blocker for this with newer versions of setuptools.

In following https://github.com/wagtail/wagtail/blob/1434a93c5418b2dbd4ec22e7ba2761c62f6fcb51/docs/contributing/developing.md#running-only-some-of-the-tests - I get:

```
$ tox -e py39-dj32-sqlite-noelasticsearch wagtail
py39-dj32-sqlite-noelasticsearch create: /Users/tomkins/Python/wagtail/.tox/py39-dj32-sqlite-noelasticsearch
py39-dj32-sqlite-noelasticsearch installdeps: django-sendfile==0.3.6, Embedly, Django~=3.2.0
ERROR: invocation failed (exit code 1), logfile: /Users/tomkins/Python/wagtail/.tox/py39-dj32-sqlite-noelasticsearch/log/py39-dj32-sqlite-noelasticsearch-1.log
=========================================================================== log start ============================================================================
Looking in indexes: https://pypi.org/simple
Obtaining file:///Users/tomkins/Python/wagtail
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting django-sendfile==0.3.6
  Using cached django-sendfile-0.3.6.tar.gz (6.9 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting Embedly
  Using cached Embedly-0.5.0.tar.gz (7.8 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [2 lines of output]
      Warning: 'classifiers' should be a list, got type 'tuple'
      error in Embedly setup command: use_2to3 is invalid.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: pip install --upgrade pip

============================================================================ log end =============================================================================
ERROR: could not install deps [django-sendfile==0.3.6, Embedly, Django~=3.2.0]; v = InvocationError("/Users/tomkins/Python/wagtail/.tox/py39-dj32-sqlite-noelasticsearch/bin/pip install -e '.[testing]' -U django-sendfile==0.3.6 Embedly 'Django~=3.2.0'", 1)
____________________________________________________________________________ summary _____________________________________________________________________________
ERROR:   py39-dj32-sqlite-noelasticsearch: could not install deps [django-sendfile==0.3.6, Embedly, Django~=3.2.0]; v = InvocationError("/Users/tomkins/Python/wagtail/.tox/py39-dj32-sqlite-noelasticsearch/bin/pip install -e '.[testing]' -U django-sendfile==0.3.6 Embedly 'Django~=3.2.0'", 1)
```

This has been reported at https://github.com/embedly/embedly-python/issues/31, but commits/releases are close to non-existent for the project. Looking at https://github.com/embedly/embedly-python/issues/33 - there is a fork of the package available.

So, the immediate fix is to either remove embedly from tox.ini for testing, or potentially use the fork. There's currently no consequences to the test suite for failures - as all of the tests check if embedly is installed first. I'm slightly hesistant in switching this to a fork - there's no guarantee that it'll be maintained in the long run.

As a follow up question to this - is anyone actually using embedly and should Wagtail continue to support it?

lb- commented:
Thanks @tomkins for the detailed description. I've flagged as needs design decision so core team can discuss this a bit.

lb- reviewed (changes requested):
@tomkins after a quick chat with the team it seems this change makes sense. 

However, we should also add Embedly to the testing extras requirements

https://github.com/wagtail/wagtail/blob/3dcde23e02e7f343fa93729ea1ade6b06b1f760c/setup.py#L42

Otherwise these tests may never run in the CI.

lb- commented:
@tomkins  are you still up for actioning the feedback above?

lb- commented:
Thanks. No rush - just seeing where it's at.

tomkins commented:
Done the following:
- Moved embedly install from tox.ini to setup.py in testing_extras
- Switched to the embedly2 fork, as [embedly is unmaintained](https://github.com/embedly/embedly-python/issues/33)
- Pinned the embedly2 fork to a specific version as it's from a third party (could switch this to `>=` if preferred - let me know)
- Updated documentation to mention embedly2 rather than embedly
- Removed django-sendfile from tox.ini, the next tox issue was various sendfile tests failing, #7986 suggests switching to django-sendfile2 - but that can be done in a separate PR (might have a look at some point, currently focusing on some other fixes I've got lined up)

Anyway, to test:

```
pip install tox
tox -e py312-dj42-sqlite-noelasticsearch-customuser-tz
```

Test at least one environment to confirm that tox is working once again.

tomkins commented:
Also - I gave Embedly a quick test locally (never used it on any projects!), and it still seems to work.

thibaudcolas commented:
Thank you for working on this @tomkins! The changes here make sense to me but I do also question whether anyone uses Embedly and whether Wagtail should keep supporting it. That fork hasn’t had any update since being set up and released once 2.5 years ago.

I’ve tagged this as "needs design decision" and added more details in #828.
